lib/ProxyFactory.php

```diff
@@ -30,35 +30,27 @@ class ProxyFactory
     private $proxyFactory;
 
     /**
-     * @var EventDispatcherInterface
+     * @var MetadataFactoryInterface
      */
-    private $dispatcher;
+    private $metadataFactory;
 
     /**
-     * @var DocumentRegistry
+     * @var DocumentManagerInterface
      */
-    private $registry;
-
-    /**
-     * @var MetadataFactoryInterface
-     */
-    private $metadataFactory;
+    private $documentManager;
 
     /**
      * @param LazyLoadingGhostFactory $proxyFactory
      * @param EventDispatcherInterface $dispatcher
-     * @param DocumentRegistry $registry
      * @param MetadataFactoryInterface $metadataFactory
      */
     public function __construct(
+        DocumentManagerInterface $documentManager,
         LazyLoadingGhostFactory $proxyFactory,
-        EventDispatcherInterface $dispatcher,
-        DocumentRegistry $registry,
         MetadataFactoryInterface $metadataFactory
     ) {
+        $this->documentManager = $documentManager;
         $this->proxyFactory = $proxyFactory;
-        $this->dispatcher = $dispatcher;
-        $this->registry = $registry;
```

Comment:
Why not keeping this variable here, but instantiate it with `$this->documentManager->getRegistry()`? Would result in less changes and methods calls in this file.